semgrep-core/src/reporting/Report.ml

```diff
@@ -352,7 +353,12 @@ let make_final_result results rules ~rules_parse_time =
   let extra =
     let mk_profiling () =
       let file_times = results |> Common.map get_profiling in
-      { rules; rules_parse_time; file_times }
+      {
+        rules;
+        rules_parse_time;
+        file_times;
+        max_memory_bytes = (Gc.quick_stat ()).top_heap_words;
```

Comment:
To get bytes multiply by `Sys.word_size`

Reply:
(If you don't do that change it from `max_memory_bytes` to `max_memory_words` or at the very least pretend it's also a 64-bit machine and multiply by 8)